### DISCUSSION TITLE
Large titles not initially large

### DISCUSSION THREAD

**Discussion by thekiwi** [software-mansion/react-native-screens#1004] [Q&A] [answered]
I've spent some time looking through previous issues/discussions but not found any answers yet, so thought I'd ask.

I'm using `createNativeStackNavigator()` to create a stack navigator. I would like the initial Home Screen to have a large title when first loaded. Here's what I put:

```JSX
enableScreens(true);
const Stack = createNativeStackNavigator();

function HomeScreen() {
    return (
        <ScrollView contentInsetAdjustmentBehavior="automatic">
            <Text>Hello World</Text>
        </ScrollView>
    );
}

function App() {
    return (
        <NavigationContainer>
            <Stack.Navigator screenOptions={{ headerLargeTitle: true }}>
                <Stack.Screen
                    name="Home"
                    options={{ title: 'My Lists', headerLargeTitle: true }}
                    component={HomeScreen}
                />
                <Stack.Screen
                    name="Details"
                    component={ListDetailsScreen}
                    options={{ headerLargeTitle: true }}
                />
            </Stack.Navigator>
        </NavigationContainer>
    );
}
```

However, every time I start the app, the navigation bar starts off in small form:

<img width="394" alt="image" src="https://user-images.githubusercontent.com/710204/124836007-284bea00-df7a-11eb-896e-c951c8149692.png">

If I drag down, it works as expected, showing the large title:

<img width="393" alt="image" src="https://user-images.githubusercontent.com/710204/124836049-3ac62380-df7a-11eb-8636-98eb6cb4d0b9.png">

But I can't figure out how to get the large title to show initially - how might I achieve that?

**Comment by cabiad** [comment]
Not a core dev, but I think #645 and #649 are related to this. It also looks like there's an open PR to fix this in #670 which would remove the need for this workaround when it gets merged and released. There's a [zero timeout workaround](https://github.com/software-mansion/react-native-screens/issues/645#issuecomment-710033686) suggested on #645 which I use successfully for now.

If it helps, here's a sample implementation of the workaround and how you might use it in a screen component.

```js
function useLargeTitleWorkaround() {
  const [largeTitleWorkaroundReady, setLargeTitleWorkaroundReady] = useState(
    false,
  );

  useEffect(() => {
    const timer = setTimeout(() => setLargeTitleWorkaroundReady(true), 0);
    return () => clearTimeout(timer);
  }, []);

  return largeTitleWorkaroundReady;
}

function SomeScreen() {
  const largeTitleWorkaroundReady = useLargeTitleWorkaround();
  // (... other hook calls)

  if (!largeTitleWorkaroundReady) {
    return null;
  }

  // (.. rest of screen component implementation)
}

**Reply by thekiwi** [reply]
Ah amazing. I've now implemented the workaround you suggested and my titles now appear large on startup. 

Hopefully it gets fixed in the core, but for now, this will do me perfectly :) 

Many thanks @cabiad!

**Comment by tboba** [comment, marked as answer]
Hi @thekiwi, I see that you've already have your question answered, but let me comment that discussion, as I believe there's already a solution in the newest version of react-native-screens 😄 

I believe in the first screen `headerLargeTitle` should work straight away. If it's still not working, try to set `headerTranslucent` prop and make sure that your ScrollView has `contentInsetAdjustmentBehavior="automatic"` property set (which in your case it's already there).

Cheers!
